internal/jobservice/repository/sql_job_service.go

```diff
@@ -124,23 +143,22 @@ func jobStateStrToJSRState(jobState string) (js.JobServiceResponse_State, error)
 }
 
 // Update database with JobTable.
-func (s *SQLJobService) UpdateJobServiceDb(jobTable *JobStatus) {
+func (s *SQLJobService) UpdateJobServiceDb(jobTable *JobStatus) error {
+	// SQLite only allows one write at a time. Therefore we must serialize
+	// writes in order to avoid SQL_BUSY errors.
+	s.writeLock.Lock()
+	defer s.writeLock.Unlock()
+
 	stmt, err := s.db.Prepare("INSERT OR REPLACE INTO jobservice VALUES (?, ?, ?, ?, ?, ?)")
 	if err != nil {
 		panic(err)
```

Comment:
Let's remove this panic also.  Maybe just return this err?

Reply:
SGTM. Addressed in 262bad8.